src/ORM/Association.php

```diff
@@ -629,9 +629,7 @@ public function deleteAll($conditions)
      */
     protected function _dispatchBeforeFind($query)
     {
-        $table = $this->target();
-        $options = $query->getOptions();
-        $table->dispatchEvent('Model.beforeFind', [$query, new \ArrayObject($options), false]);
+        $query->triggerBeforeFind();
```

Comment:
I guess my worry with this change is that the target on the triggered event will be incorrect.